test/test.js

```diff
@@ -99,99 +51,89 @@ describe('Azure Swagger Schema Validation', function () {
     });
   }).value();
 
-  _(compositeSwaggers).each(function (compositeSwagger) {
-    it('composite: ' + compositeSwagger + ' should be a valid Composite Swagger document.', function (done) {
-      fs.readFile(compositeSwagger, 'utf8', function (err, data) {
-        if (err) { done(err); }
-        var parsedData;
-        try {
-          parsedData = JSON.parse(stripBOM(data));
-        } catch (err) {
-          throw new Error("compositeSwagger " + compositeSwagger + " is an invalid JSON. " + util.inspect(err, { depth: null }));
-        }
-        var valid = validator.validate(parsedData, compositeSchema);
-        if (!valid) {
-          var error = validator.getLastErrors();
-          throw new Error("Schema validation for Composite Swagger failed: " + util.inspect(error, { depth: null }));
-        }
-        assert(valid === true);
-        var compositeSwaggerDir = path.dirname(compositeSwagger);
-        var messages = [];
-        if (parsedData.documents && parsedData.documents.length > 0) {
-          async.eachSeries(parsedData.documents, function (docUrl, loopCallback) {
-            //construct the absolue path if the item in the documents array is a relative path
-            if (!path.isAbsolute(docUrl) && !docUrl.startsWith('http')) {
-              docUrl = path.join(compositeSwaggerDir, docUrl);
-            }
-            //make a request if it is a url
-            if (docUrl.startsWith('http')) {
-              request({ url: docUrl, json: true }, function (error, response, responseBody) {
-                if (error) {
-                  messages.push('An error occurred while accessing the swagger doc ' +
-                    docUrl + ' from the documents list. The error is ' + util.inspect(error, { depth: null }));
-                }
-                if (response.statusCode !== 200) {
-                  messages.push('\'' + response.statusCode + '\': \'File Not Found\'- error occurred while accessing the swagger doc ' +
-                    docUrl + ' from the documents list.');
+  describe('Azure composite swagger schema validation:', function () {
+    _(utils.compositeSwaggers).each(function (compositeSwagger) {
+      it('composite: ' + compositeSwagger + ' should be a valid Composite Swagger document.', function (done) {
+        utils.parseJsonFromFile(compositeSwagger, function (err, parsedData) {
+          if (err) { done(err); }
+          var valid = context.validator.validate(parsedData, context.compositeSchema);
+          if (!valid) {
+            var error = context.validator.getLastErrors();
+            throw new Error("Schema validation for Composite Swagger failed: " + util.inspect(error, { depth: null }));
+          }
+          assert(valid === true);
```

Comment:
What are the possible values of `valid`? I am asking because i was thinking whether we can remove this line or not?